Add a readme that contains the new style sheet (#41)
dandi/handbook: README.md

# handbook
Handbook for interacting with the DANDI archive.

> ## DANDI Style Sheet

Author: That's a fair point. I used style "sheet" instead of "guide" because (for example) we have a comprehensive style guide at Kitware, and I used "sheet" to distinguish that from this more project-specific set of guidelines. I will rename it to "DANDI Style Guidelines".

Follow the guidelines below when creating and revising text in the DANDI Handbook:

> *	**dandi-** repositories (hyphenate, dandi-archive, dandi-cli, *not* dandischema; Dandisets is an exception because that is a

Reviewer: ```suggestion
*	The names of most DANDI GitHub repositories are hyphenated (dandi-archive, dandi-cli; does not apply to dandischema); "Dandisets" is an exception because that is a
```

Reviewer 2: @jwodder, I think the idea here was to standardize even `dandischema`. Is that one excepted from the hyphenation pattern because it is associated with a Python package?

Author: I also noticed I had an extra comma in here, I will rewrite to clarify, thanks.

Reviewer 2: > It's excepted from hyphenation because the name of the repository isn't hyphenated. Either way, the item needs to be rewritten to be clearer about dandischema.

I am sure you realize the circularity of your statement. Repository names can be changed to match up with a style guide, or style guides can be changed to except specific repository names--but both of those plans would require some justification.

My question is whether the repository name *needs* to remain unhyphenated for some other reason (such as wanting to keep it the same as the resulting Python package). If not, I'm proposing that we follow the style guide and change the name of the repo.

Author: @jwodder & @waxlamp - I am thinking I should revise as follow:

```suggestion
*	DANDI repository names - follow the spelling that is used in GitHub (dandi-archive and dandi-cli are hyphenated, dandischema and dandisets are not hyphenated)
```

Author: > @waxlamp - i don't think anything prevents us from hyphenating dandischema. the library name can be decoupled from the repo name. there may be some files we may need to modify if some test or dockerfile directly references the reponame. github, i believe, offers redirects from old names for a while.

@waslamp - if @satra is ok with changing the repo names to match the convention (i.e. changing "dandischema" to  "dandi-schema") then I will revise to say 
```suggestion
*	**dandi-** repository names should be hyphenated (e.g. dandi-archive, dandi-cli, dandischema); Dandisets is an exception because that is a
```

Reviewer 2: Let's go ahead and change it.

@melster1010, in your suggested change, did you mean to change `dandischema` to `dandi-schema`?

> *	**dandi-** repositories - hyphenate the names of DANDI GitHub repositories (e.g. **dandi-archive**); Dandisets is an exception because it is a

Reviewer: ```suggestion
*	**dandi-** repositories - Hyphenate the names of DANDI GitHub repositories (e.g. **dandi-archive**); "Dandisets" is an exception because it is a
```

Author: ```suggestion
*	**dandi-** repositories - hyphenate the names of DANDI GitHub repositories (e.g. **dandi-archive**); "Dandisets" is an exception because it is a
```

Author: these are bulleted items not full sentences so I'm not capitalizing the first word, will add quotes to Dandisets though

complete word, i.e. it is not using dandi as a prefix to a noun)

> *	**dandiset** (**not** Dandiset)

Reviewer 3: ```suggestion
*	**Dandiset** (**not** dandiset)
```

> *	**dandiset** (**not** Dandiset)

Reviewer 3: i'll need to think a bit about this one. i think this should be capitalized.

Author: I'm fine with consistently capitalizing "Dandiset"

Author: now it's  **Dandiset** (**not** dandiset)`

*	file names - use lower case (e.g. **development.md**)
*	headings - use Title Capitalization (for 1st and 2nd levels) and follow with an intro sentence

> *	**GitHub** camel case (**not** github or Github)

Reviewer: ```suggestion
*	**GitHub** — use camel case (**not** github or Github)
```

> *	instructional language should be direct, imperative, active, straightforward (e.g. "Install the files in your Python environment" *not* "Files could be installed in your Python environment")

Reviewer: ```suggestion
*	Instructional language should be direct, imperative, active, straightforward (e.g. "Install the files in your Python environment", *not* "Files could be installed in your Python environment")
```

Author: not capitalizing the first word in these bulleted phrases

Author: ```suggestion
*	instructional language should be direct, imperative, active, straightforward (e.g. "Install the files in your Python environment", *not* "Files could be installed in your Python environment")
```

> *	**JupyterHub** camel case (**not** Jupyterhub)

Reviewer: ```suggestion
*	**JupyterHub** — use camel case (**not** Jupyterhub)
```
